### DISCUSSION TITLE
Allow using 'include' function in any gotmpl file in a convenient way

### DISCUSSION THREAD

**Discussion by Vince-Chenal** [helmfile/helmfile#1271] [Ideas]
Hello @yxxhero 

I've been playing with the `include` function [you recently introduced](https://github.com/helmfile/helmfile/pull/1187) and noticed something that could be improved.

I wanted to use `include` function from another file than `helmfile.yaml`. For it to work I had to place my `_helpers.tpl` next to my values file like this:

```
.
├── helmfile.yaml
└── values
    ├── _helpers.tpl
    └── myvalues.yaml.gotmpl
```

Otherwise the helper function was not found:
```
in ./helmfile.yaml: failed to render values files "values/myvalues.yaml.gotmpl": failed to render [values/myvalues.yaml.gotmpl], because of template: stringTemplate:18:14: executing "stringTemplate" at <include "myhelper" .>: error calling include: template: no template "myhelper" associated with template "stringTemplate"
```

With for example:

helmfile.yaml:
```
---
releases:
  - name: myrelease
    chart: oci://...mychart
    version: 0.0.1
    values:
      - values/myvalues.yaml.gotmpl
```

and  `myvalues.yaml.gotmpl`:
```
---
somevalue: {{ include "myhelper" . }}
```

It would be great if helpers were always loaded from the root path (next to `helmfile.yaml`), otherwise I would be forced to copy paste my helper functions everywhere I need them which goes against the concept of re-usability of helpers

**Comment by yxxhero** [comment]
see this: https://github.com/helmfile/helmfile/pull/1187#discussion_r1412908641

@mumoshu WDYT? this is a true use case.

**Reply by yxxhero** [reply]
@Vince-Chenal you can try to use link file for _helper.tpl.

**Reply by Vince-Chenal** [reply]
I tried, it works, but same thing I don't want to have links all around in my repo. Wouldn't it make sense to have a common place for helpers? At least common to one helmfile.

I understand the nested helmfile case raises other questions like the helpers scope:
- Should parent helmfile have access to subhelmfile helpers?
- Should subhelmfile have access to parent helpers?
- If yes, who wins if both helmfiles create two helper functions with same name?

For example, with helm chart dependencies and [named templates](https://helm.sh/docs/chart_template_guide/named_templates/):
> An important detail to keep in mind when naming templates: template names are global. If you declare two templates with the same name, whichever one is loaded last will be the one used. Because templates in subcharts are compiled together with top-level templates, you should be careful to name your templates with chart-specific names.
